highfidelity/hifi
# Reworked Particle Editor UI

Whats new

- Rewrote particle editor.
- Particle Editor style now matches the rest of the edit
- Sliders for Float values. Sliders are limited to a max value as defined, but the max value can be increased by setting the input field larger than max. Slider values should translate as soon as you stop moving the slider around
- Textures now are previewed inside the editor after url has been changed
- Colors now bind nearly immediately without having to press ok on the particle color picker.
- Azimuth, Polar values are now set using degrees instead of radians (does the conversion in the background for ease of use)
- Added Some Magic UI Sauce to make things feel abit more polished for the particle editor.
- Removed DatGUI
- Updated underscore
- Created an UI system based on High Fidelity's style for Edit/Create, can later be expanded further to other parts of edit.

Stuff to test (both desktop and hmd):
- See that sections are the same [as in spec](https://worklist-production.s3.amazonaws.com/attachment/0aea9e6d-2288-d99b.pdf) and @AlanZimmerman feedback below.. Note that tab dissapearing and non-dissapearing will not be according to spec due to [tech issues with qml](https://www.youtube.com/watch?v=bKjODpV8lu8&feature=youtu.be). 
- Stylisticly same or similar as rest of edit.
- See as to particle system working the same as it used to: If non-particle entity is selected
- All but the Radian sliders (Azimuth Polar), and float values with max value of 1, should be overridable by typing in a larger value instead of using slider, this should allow for the slider to be moved around. When this is done, sliders should update their max value.
- Colors should update immediately, without having to press OK
- Show Properties should toggle properties,
- Show Properties should enable export button
- Show Properties should not enabled import button
- Changing json properties should enable import button
- When import button is pressed, all fields should update to same as what was in the properties. If field is missing it is skipped
- When Export button is pressed, it should highlight the properties field (and maybe add it to clipboard, depending on support)
